extractor/service.go

```diff
@@ -1,20 +1,240 @@
+func (s *Service) GetRepositoryData(r *Request) (*RepositoryData, error) {
 	// TODO
 	return nil, fmt.Errorf("NOT IMPLEMENTED YET")
 }
 
 //proteus:generate
-func (*Service) GetRepositoriesData() ([]*RepositoryData, error) {
-	return nil, fmt.Errorf("NOT IMPLEMENTED YET")
+func (s *Service) GetRepositoriesData() ([]*RepositoryData, error) {
+	n, err := core.ModelRepositoryStore().Count(model.NewRepositoryQuery().FindByStatus(model.Fetched))
+	checkIfError(err)
+	fmt.Printf("Iterating over %d 'fetched' repositories in DB:\n", n)
+
+	master := "refs/heads/master"
+	result := make([]*RepositoryData, n)
+
+	reposNum := 0
+	totalFilesNum := 0
+	for masterRefInit, repoID := range findAllFetchedReposWithRef(master) {
+		fmt.Printf("Repo: %v", repoID)
+		repo := &RepositoryData{
+			RepositoryID: repoID,
+			URL:          "", //TODO(bzz): add repo url!
+			Files:        make([]File, 100),
+		}
+
+		rootedTransactioner := core_retrieval.RootedTransactioner()
+		tx, err := rootedTransactioner.Begin(plumbing.Hash(masterRefInit))
+		if err != nil {
+			fmt.Printf("Failed to begin a tx for repo:%v, hash:%v. %v\n", repoID, masterRefInit, err)
+			continue
+		}
+
+		tree, err := gitOpenGetTree(tx.Storer(), repoID, masterRefInit, master)
+		if err != nil {
+			fmt.Printf("Failed to open&get tree for Git repo:%v, hash:%v. %v\n", repoID, masterRefInit, err)
+			_ = tx.Rollback()
+			continue
+		}
+
+		skpFilesNum := 0
+		sucFilesNum := 0
+		errFilesNum := 0
+		err = tree.Files().ForEach(func(f *object.File) error {
+			i := (skpFilesNum + sucFilesNum + errFilesNum) % 1000
+			batch := (skpFilesNum + sucFilesNum + errFilesNum) / 1000
+			if i == 0 && batch != 0 {
+				fmt.Printf("\t%d000 files...\n", batch)
+			}
+
+			// discard vendoring with enry
+			if enry.IsVendor(f.Name) || enry.IsDotFile(f.Name) ||
+				enry.IsDocumentation(f.Name) || enry.IsConfiguration(f.Name) {
+				skpFilesNum++
+				return nil
+			} //TODO(bzz): filter binaries like .apk and .jar
+
+			// detect language with enry
+			fContent, err := f.Contents()
+			if err != nil {
+				fmt.Printf("\tFailed to read file %s, %s\n", f.Name, err)
+				errFilesNum++
+				return nil
+			}
+
+			fLang := enry.GetLanguage(f.Name, []byte(fContent))
+			if err != nil {
+				fmt.Printf("\tFailed to detect language for %s, %s\n", f.Name, err)
+				errFilesNum++
+				return nil
+			}
+			//fmt.Printf("\t%-9s blob %s    %s\n", fLang, f.Hash, f.Name)
+
+			// Babelfish -> UAST (Python, Java)
+			if strings.EqualFold(fLang, "java") || strings.EqualFold(fLang, "python") {
+
+				//TODO(bzz): reply as protobuf
+				uast, err := parseToUast(s.bblfshClient, f.Name, strings.ToLower(fLang), fContent)
+				if err != nil {
+					errFilesNum++
+					return nil
+				}
+
+				sucFilesNum++
+				file := File{
+					Language: fLang,
+					Path:     f.Name,
+					UAST:     string(*uast),
+				}
+				repo.Files = append(repo.Files, file)
+			}
+			return nil
+		})
+		checkIfError(err)
```

Comment:
We are skipping all failing files and handing all errors inside `FileIter.ForEach`